packages/docusaurus-theme-classic/src/theme/NavbarItem/index.tsx

```diff
@@ -43,18 +43,23 @@ const getNavbarItemComponent = (type: NavbarItemComponentType) => {
   return navbarItemComponentFn();
 };
 
-function getComponentType({type, ...props}: Props): NavbarItemComponentType {
+function getComponentType(
+  type: string | undefined,
```

Comment:
The type can actually be more restricted:

```suggestion
  type: Types,
```